yt_dlp/extractor/abematv.py

```diff
@@ -412,6 +414,7 @@ def _real_extract(self, url):
         info.update({
             'id': video_id,
             'title': title,
+            'thumbnails': [{'url': thumbnail}],
```

Comment:
Does `self._og_search_thumbnail` not work?
```suggestion
            'thumbnail': self._og_search_thumbnail(webpage),
```

Reply:
self._og_search_thumbnail can work, but the thumbnail obtained this way is also the lowest resolution. You need to remove all parameters from the URL to get the original resolution webp thumbnail.

Reply:
```suggestion
            'thumbnail': update_url(self._og_search_thumbnail(webpage) or '', query=None) or None,
```
And import `update_url` from `..utils`

Reply:
OK, this looks very good, but some videos can get thumbnails. If assigned to thumbnail, the url in thumbnails will still be downloaded, resulting in the lowest resolution download. I think this needs to be changed.

```suggestion
            'thumbnails': [{'url': update_url(self._og_search_thumbnail(webpage) or '', query=None) or None}],
```


![PixPin_2025-04-07_16-08-34](https://github.com/user-attachments/assets/e716e2e7-f7d8-4746-b111-fb74d17ef627)
![PixPin_2025-04-07_16-11-08](https://github.com/user-attachments/assets/5284f97d-c1c7-4726-91cc-cb9a85e5ca3c)

Reply:
yes (for /episode URLs)

ABEMA code is outdated and mainly retrieves information from webpage, which means some URL formats (like /slots ...) are only partially supported
so I think this temporary solution is acceptable

now that information can be fetched via api, I'll rewrite the entire code soon

Reply:
```
        if thumbnail := update_url(self._og_search_thumbnail(webpage, default=''), query=None):
            info['thumbnails'] = [{'url': thumbnail}]
```
My requirement is that when I manually get the thumbnail, I will use it, regardless of whether `info['thumbnails']` has a value or not, I will discard it, so I think it should be like this

Reply:
why discard if we can just return multiple thumbs??

Reply:
![PixPin_2025-04-08_09-38-17](https://github.com/user-attachments/assets/de1d9edf-9407-4107-a383-568f16b42a02)

```
[AbemaTV] Extracting URL: https://abema.tv/video/episode/90-2027_s1_p603
[AbemaTV] Authorizing
[AbemaTV] 90-2027_s1_p603: Downloading webpage
[AbemaTV] 90-2027_s1_p603: Checking playability
[AbemaTV] 90-2027_s1_p603: Downloading m3u8 information
[info] 90-2027_s1_p603: Downloading 1 format(s): 5300
[info] Downloading video thumbnail 1 ... https://image.p-c2-x.abema-tv.com/image/programs/90-2027_s1_p603/thumb001.png?height=144&quality=75&version=1743737614&width=256
[info] Writing video thumbnail 1 to: FRUITS ZIPPER卒業生に贈る言葉＆先生とコラボで最後のサプライズ [90-2027_s1_p603].png
```

The difference between these two URLs lies in the parameters appended. I want to obtain images in their original resolution, but the presence of parameters reduces the image quality. However, yt-dlp downloads URLs with parameters during execution, which appears to be caused by its internal logic.

Reply:
![PixPin_2025-04-08_09-59-15](https://github.com/user-attachments/assets/5c872311-b579-4d41-8472-bc218b811754)
The code `[::-1]` in line 4407 causes the array to be reversed in order.